dtyper.py

```diff
@@ -34,39 +50,36 @@ def dataclass_maker(function_or_class):
     return dataclass_maker
 
 
-def function(typer_command):
-    params = list(_params(typer_command))
+def function(typer_command: Callable[P, R]) -> Callable[P, R]:
+    """Return function that can be called outside of a typer.Typer app context
+
+    This allows a function with default argument values of instance `typer.Option`
+    and `typer.Argument` to be called without having to provide all the defaults
+    manually.
+    """
+
+    new_sig = _fixed_signature(typer_command)
 
     @wraps(typer_command)
-    def wrapped(*args, **kwargs):
-        tc = typer_command.__name__ + '()'
-
-        if len(args) > len(params):
-            lp = len(params)
-            s = 's' * (lp != 1)
-            la = len(args)
-            raise TypeError(f'{tc} takes {lp} argument{s} but {la} were given')
-
-        for name, _, default in params[len(args):]:
-            value = kwargs.pop(name, default)
-            if value is ...:
-                raise TypeError(f'{tc} missing required parameter \'{name}\'')
-            args = *args, value
-        return typer_command(*args, **kwargs)
+    def wrapped(*args: P.args, **kwargs: P.kwargs) -> R:
+        # https://docs.python.org/3/library/inspect.html#inspect.Signature.bind
```

Comment:
hah, that would have saved me time!

Still, delete that comment, now you've exposed this better way.  :-)